python/wpt/test.py

```diff
@@ -50,48 +50,86 @@
 
 @dataclasses.dataclass
 class MockPullRequest():
+    """
+    A mock representation of a GitHub pull request.
+
+    Attributes:
+        head (str): The name of the branch where changes are made.
+        number (int): The pull request number.
+        state (str): The state of the pull request (default is "open")
+    """
     head: str
     number: int
     state: str = "open"
 
 
 class MockGitHubAPIServer():
+    """
+    A mock server to simulate GitHub API for testing purposes.
+    """
     def __init__(self, port: int):
         self.port = port
         self.disable_logging()
         self.app = flask.Flask(__name__)
         self.pulls: list[MockPullRequest] = []
 
-        class NoLoggingHandler(WSGIRequestHandler):
-            def log_message(self, *args):
-                pass
-        if logging.getLogger().level == logging.DEBUG:
-            handler = WSGIRequestHandler
-        else:
-            handler = NoLoggingHandler
-
-        self.server = make_server('localhost', self.port, self.app, handler_class=handler)
+        def __init__(self, port: int):
+        self.port = port
+        self.disable_logging()
+        self.app = flask.Flask(__name__)
+        self.pulls: List[MockPullRequest] = []
+        self.configure_routes()
         self.start_server_thread()
 
     def disable_logging(self):
+        """
+        Disables Flask and Werkzeug logging to keep the output clean.
+        """
         flask.cli.show_server_banner = lambda *args: None
-        logging.getLogger("werkzeug").disabled = True
-        logging.getLogger('werkzeug').setLevel(logging.CRITICAL)
+        log = logging.getLogger('werkzeug')
+        log.setLevel(logging.CRITICAL)
+        log.disabled = True
+
+    def configure_routes(self):
+        """
+        Configures all the routes for the mock GitHub API server.
+        """
+        self.app.add_url_rule("/ping", view_func=self.ping)
+        self.app.add_url_rule("/reset-mock-github", view_func=self.reset_server, methods=['GET'])
+        self.app.add_url_rule("/repos/<org>/<repo>/pulls/<int:number>/merge", view_func=self.merge_pull_request, methods=['PUT'])
+        self.app.add_url_rule("/search/issues", view_func=self.search, methods=['GET'])
+        self.app.add_url_rule("/repos/<org>/<repo>/pulls", view_func=self.create_pull_request, methods=['POST'])
+        self.app.add_url_rule("/repos/<org>/<repo>/pulls/<int:number>", view_func=self.update_pull_request, methods=['PATCH'])
+        self.app.add_url_rule("/repos/<org>/<repo>/issues/<number>/labels", view_func=self.other_requests, methods=['GET', 'POST'])
+        self.app.add_url_rule("/repos/<org>/<repo>/issues/<number>/labels/<label>", view_func=self.other_requests, methods=['DELETE'])
+        self.app.add_url_rule("/repos/<org>/<repo>/issues/<issue>/comments", view_func=self.other_requests, methods=['GET', 'POST'])
 
```

Comment:
I think using the decorators on the functions is a bit nicer. Can you explain why you have chosen to move away from the decorator approach?